qiskit/circuit/register.py

```diff
@@ -84,6 +84,8 @@ def name(self):
     @name.setter
     def name(self, value):
         """Set the register name."""
+        if self.name_format.match(value) is None:
+            raise CircuitError("%s is an invalid OPENQASM register name." % value)
```

Comment:
Could we add a reference where users could look up valid names?

Reply:
Thanks for the suggestion @Cryoris . I have added the link to the Open QASM arXiv document to the error message. I have also mentioned it in the release note.